tests/audio/test_sdr.py

```diff
@@ -0,0 +1,154 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from collections import namedtuple
+from functools import partial
+
+import pytest
+import torch
+from mir_eval.separation import bss_eval_sources
+from torch import Tensor
+
+from tests.helpers import seed_all
+from tests.helpers.testers import MetricTester
+from torchmetrics.audio import SDR
+from torchmetrics.functional import sdr
+from torchmetrics.utilities.imports import _TORCH_GREATER_EQUAL_1_6
+
+seed_all(42)
+
+Input = namedtuple("Input", ["preds", "target"])
+
+inputs_1spk = Input(
+    preds=torch.rand(3, 2, 1, 8000),
+    target=torch.rand(3, 2, 1, 8000),
+)
+inputs_2spk = Input(
+    preds=torch.rand(3, 2, 2, 8000),
+    target=torch.rand(3, 2, 2, 8000),
+)
```

Comment:
We are using 2 devices for ddp testing, which both requires the same number of batches to process (to prevent a dead-lock due to work imbalance). So number of batches need to be divisible by 2.
```suggestion
inputs_1spk = Input(
    preds=torch.rand(4, 2, 1, 8000),
    target=torch.rand(4, 2, 1, 8000),
)
inputs_2spk = Input(
    preds=torch.rand(4, 2, 2, 8000),
    target=torch.rand(4, 2, 2, 8000),
)
```